OpenRA.Mods.Common/Widgets/ConfirmationDialogs.cs

```diff
@@ -37,6 +49,23 @@ public static void PromptConfirmAction(string title, string text, Action onConfi
 				if (onCancel != null)
 					onCancel();
 			};
+
+			var restartButton = prompt.Get<ButtonWidget>("RESTART_BUTTON");
+			if (world != null)
+			{
+				var isVisible = world.IsReplay || world.Map.Visibility == MapVisibility.MissionSelector; // only show for missions and replays
```

Comment:
This check should be done by the code calling `PromptConfirmAction` instead of hardcoding it here.